pkg/engine/engine_test.go

```diff
@@ -288,19 +290,19 @@ global:
 		t.Fatalf("failed to render templates: %s", err)
 	}
 
-	if out[outerpath] != "Gather ye rosebuds while ye may" {
+	if out["top/"+outerpath] != "Gather ye rosebuds while ye may" {
 		t.Errorf("Unexpected outer: %q", out[outerpath])
 	}
 
-	if out[innerpath] != "Old time is still a-flyin'" {
+	if out["top/charts/herrick/"+innerpath] != "Old time is still a-flyin'" {
 		t.Errorf("Unexpected inner: %q", out[innerpath])
 	}
 
-	if out[deepestpath] != "And this same flower that smiles to-day" {
+	if out["top/charts/herrick/charts/deepest/"+deepestpath] != "And this same flower that smiles to-day" {
```

Comment:
Perhaps `deepestpath` could be named `inner.tpl` or `outer.tpl` to have a test that ensures collisions don't occur? Or it could even be a separate test.